### PR TITLE
[NON-MODULAR] Removes Final Objectives

### PR DESCRIPTION
**Pull request by axietheaxolotl**
## About The Pull Request

Read the tin.

## How This Contributes To The Skyrat Roleplay Experience

Spawning 13 nukies after doing two objectives isn't okay, and Romerol is just bad.

Also, Xyel wanted this.

## Changelog

:cl:
del: removed final objectives
/:cl:

### CODE REVIEW

**Comment by Xymec** [comment]
can't have shit on skyrat. lmao, I get with the 13 nukies and the romerol though.

**Comment by KathyRyals** [comment]
Feels like we should just remove the current ones and implement our own, not remove the final objective system wholesale.

**Comment by axietheaxolotl** [comment]
> Feels like we should just remove the current ones and implement our own, not remove the final objective system wholesale.

When new ones get added, we'll reimplement the system.

File: code/modules/antagonists/traitor/uplink_handler.dm [hunk 1]
```diff
@@ -179,8 +179,8 @@
 		objective.update_progression_reward()
 
 /datum/uplink_handler/proc/abort_objective(datum/traitor_objective/to_abort)
-	if(istype(to_abort, /datum/traitor_objective/final))
-		return
+//	if(istype(to_abort, /datum/traitor_objective/final)) // SKYRAT EDIT REMOVAL
+//		return // SKYRAT EDIT REMOVAL
```
Comment (KathyRyals) [inline code/modules/antagonists/traitor/uplink_handler.dm]: By that I mean, I don't see a reason to comment that out. You commented the objectives datum, commenting this as well just seems nonsensical to me.

File: code/modules/antagonists/traitor/objectives/final_objective/final_objective.dm [hunk 2]
```diff
@@ -1,3 +1,4 @@
+/* SKYRAT EDIT REMOVAL
 /datum/traitor_objective_category/final_objective
 	name = "Final Objective"
 	objectives = list(
```
Comment (KathyRyals) [inline code/modules/antagonists/traitor/objectives/final_objective/final_objective.dm]: Ditto

File: code/modules/antagonists/traitor/objectives/final_objective/final_objective.dm [hunk 3]
```diff
@@ -33,3 +34,4 @@
 /datum/traitor_objective/final/uplink_ui_data(mob/user)
 	. = ..()
 	.["final_objective"] = TRUE
+*/
```
Comment (KathyRyals) [inline code/modules/antagonists/traitor/objectives/final_objective/final_objective.dm]: Ditto

File: code/modules/antagonists/traitor/uplink_handler.dm (hunk 1 above)
Comment (axietheaxolotl) [reply code/modules/antagonists/traitor/uplink_handler.dm]: that's fair

**Comment by KathyRyals** [comment]
Changed my 👎 to a 👍

**Comment by Iamgoofball** [comment]
:-1:
You need to make replacement objectives for these before removing them.

File: code/modules/antagonists/traitor/objectives/final_objective/final_objective.dm [hunk 4]
```diff
@@ -1,8 +1,8 @@
 /datum/traitor_objective_category/final_objective
 	name = "Final Objective"
 	objectives = list(
-		/datum/traitor_objective/final/romerol = 1,
-		/datum/traitor_objective/final/battlecruiser = 1,
+//		/datum/traitor_objective/final/romerol = 1, // SKYRAT EDIT
```
Comment (Iamgoofball) [inline code/modules/antagonists/traitor/objectives/final_objective/final_objective.dm:4]: You need to actually make replacement objectives.

**Comment by axietheaxolotl** [comment]
>  You need to make replacement objectives for these before removing them.

Out of scope, this just removes current ones as they only serve as ban-bait.

**Review by tf-4 — APPROVED** [review]
Host request

File: code/modules/antagonists/traitor/objectives/final_objective/final_objective.dm [hunk 5]
```diff
@@ -1,8 +1,8 @@
 /datum/traitor_objective_category/final_objective
```
Comment (Gandalf2k15) [inline code/modules/antagonists/traitor/objectives/final_objective/final_objective.dm:1]: Undo your changes to the rest of the files and simply comment this out.

**Comment by Zenitheevee** [comment]
eh- Maybe we can end up making people have to do more than just 2 objectives in the future
but yeah a comment out seems like a good idea for now.

**Comment by Iamgoofball** [comment]
>Out of scope, this just removes current ones as they only serve as ban-bait.

This is not out of scope, you literally are ripping out part of progression traitors and not properly replacing it. Properly replace it.

**Comment by Gandalf2k15** [comment]
I meant remove the actual objs from that list thing

File: code/modules/antagonists/traitor/objectives/final_objective/final_objective.dm [hunk 6]
```diff
@@ -1,8 +1,9 @@
+/* SKYRAT EDIT
 /datum/traitor_objective_category/final_objective
 	name = "Final Objective"
 	objectives = list(
-		/datum/traitor_objective/final/romerol = 1,
```
Comment (Gandalf2k15) [inline code/modules/antagonists/traitor/objectives/final_objective/final_objective.dm:4]: Make it an empty list is what I meant

**Comment by RatFromTheJungle** [comment]
> 👎 You need to make replacement objectives for these before removing them.

you aren't a maintainer

**Comment by RatFromTheJungle** [comment]
> > Out of scope, this just removes current ones as they only serve as ban-bait.
> 
> This is not out of scope, you literally are ripping out part of progression traitors and not properly replacing it. Properly replace it.

See above

File: code/modules/antagonists/traitor/objectives/final_objective/final_objective.dm (hunk 6 above)
Comment (axietheaxolotl) [reply code/modules/antagonists/traitor/objectives/final_objective/final_objective.dm:4]: 💀

**Pull request closed without merging**